tcpclient/connectionsManagement_test.go

```diff
@@ -13,41 +12,93 @@ import (
-		t.Log("Initiating runner # ", strconv.Itoa(runner))
-		go TCPConnect(runner, host, port, &wg, ioutil.Discard, make(chan Connection, numberConnections), make(chan bool))
-		t.Logf("Runner %s initated. Remaining: %s", strconv.Itoa(runner), strconv.Itoa(numberConnections-runner))
+	var statusChannel = make(chan Connection, 2)
+	var closeRequest = make(chan bool)
+
+	t.Log("Initiating TCP Connect")
+	go TCPConnect(1, host, port, &wg, ioutil.Discard, statusChannel, closeRequest)
+	time.Sleep(1 * time.Second)
+	if (<-statusChannel).GetConnectionStatus() == ConnectionDialing {
+		t.Log("Connection Dialing")
+	} else {
+		t.Error("Connection failed to dial")
+	}
+	var connectionEstablished = <-statusChannel
+	if connectionEstablished.GetConnectionStatus() == ConnectionEstablished {
+		t.Log("Connection Estalished")
+	} else {
+		t.Error("Connection failed to establish")
 	}
+	if connectionEstablished.GetTCPProcessingDuration(ConnectionEstablished) != 0 {
+		t.Log("Connection Estalished in ", connectionEstablished.GetTCPProcessingDuration(ConnectionEstablished))
+	} else {
+		t.Error("Connection TCP Processing Duration not consistent")
+	}
+}
 
-	t.Log("Waiting runners to finish")
-	time.Sleep(time.Second)
+func TestTcpConnectErrored(t *testing.T) {
+	var host = "127.0.0.1"
+	var port = 55556
+
+	var wg sync.WaitGroup
+	wg.Add(1)
 
-	// Marking wait groups as done after a second does not make sense...
-	// and in case one client finishes before this loop being executed (because of an error?)
-	// we will get an exception (panic: sync: negative WaitGroup counter [recovered])
+	var statusChannel = make(chan Connection, 2)
+	var closeRequest = make(chan bool)
 
-	//for runner := 1; runner <= numberConnections; runner++ {
-	//	t.Log("Closing runner #", strconv.Itoa(runner))
-	//	wg.Done()
-	//}
+	t.Log("Initiating TCP Connect")
+	go TCPConnect(1, host, port, &wg, ioutil.Discard, statusChannel, closeRequest)
+	time.Sleep(1 * time.Second)
+	if (<-statusChannel).GetConnectionStatus() == ConnectionDialing {
+		t.Log("Connection Dialing")
+	} else {
+		t.Error("Connection failed to dial")
+	}
 
+	var connectionErrored = <-statusChannel
+
+	if connectionErrored.GetConnectionStatus() == ConnectionError {
+		t.Log("Connection Errored")
+	} else {
+		t.Error("Connection not errored")
+	}
+	if connectionErrored.GetTCPProcessingDuration(ConnectionError) != 0 {
+		t.Log("Connection Errored in ", connectionErrored.GetTCPProcessingDuration(ConnectionError))
+	} else {
+		t.Error("Connection TCP Processing Duration not consistent")
+	}
+}
+
+func TestReportConnectionStatus(t *testing.T) {
+	var debugOut = ioutil.Discard
+	connStatusCh := make(chan Connection, 1)
+	connectionDescription := Connection{
+		ID:      0,
+		status:  ConnectionDialing,
+		metrics: connectionMetrics{},
+	}
+	reportConnectionStatus(debugOut, connStatusCh, connectionDescription)
+	if <-connStatusCh != connectionDescription {
+		t.Log("Not proper Connection reported: ", <-connStatusCh)
```

Comment:
why don't you pass the message to the t.Error?

Reply:
my mistake